drive.go

```diff
@@ -144,15 +145,15 @@ func (ds *DriveService) ListChildren(ctx context.Context, driveID string, filter
 	return items, resp, nil
 }
 
-// Search the hierarchy of items for items matching a query
-func (ds *DriveService) Search(ctx context.Context, q string, filter string) (*Items, *http.Response, error) {
-	q = url.QueryEscape(q)
-
-	if filter != "" {
-		q = q + "&filter=" + url.QueryEscape(filter)
+// Search the hierarchy of items for all items the user has access to.
+func (ds *DriveService) Search(ctx context.Context, q string) (*Items, *http.Response, error) {
+	// We need to escape single quotes with a double single
+	// quotes, otherwise API will fail with unterminated
+	if strings.Contains(q, "'") {
+		q = strings.ReplaceAll(q, "'", "''")
```

Comment:
perhaps this could be `pathEscapeIncludingQuotes` and have it encode single quotes as well as calling `url.PathEscape` in one helper?

Reply:
great will do 👍

Reply:
@glenjamin if I understand the edit question the url encoding isn't just for the single quotes - but other special characters that would break if the user tried them and were not encoded.